### ISSUE TITLE
Jackson-databind Vulnerabilities

### ISSUE THREAD

**Issue by jermynyeo** [AbsaOSS/spline-spark-agent#547]
Hi Team, 

I'm trying to procure this package. However, it's failing SSAP scans due to [vulnerabilities](https://www.cvedetails.com/vulnerability-list/vendor_id-15866/product_id-42991/Fasterxml-Jackson-databind.html) in Jackson-databind packages.  
Could  the version of `jackson-databind` be upgraded to 2.13.3 and above?

**Comment by cerveada** [comment]
Should be possible unless it causes incompatibilities with Spark or other libraries we use.

Please not that if you use one of spark-spline-agent-bundles this dependency is marked as provided. We just use the one that is already present in Spark.

**Comment by wajda** [comment]
> ...if you use one of spark-spline-agent-bundles this dependency is marked as provided...

Just to clarify, Spline agent does not use `jackson-databind` library at all. It appears in bundles' `pom.xml` files in the _dependency management_ section, which is basically copied from Spark `pom.xml` of the respective version. It does not create any dependency on that library in the Spline agent. From this perspective the SSAP warning is false positive.

**Comment by jermynyeo** [comment]
Hi, thanks for the clarifications.  I'll close this for now and reach out again if I face any issues. Thanks!

**Comment by jermynyeo** [comment]
Hello, I just checked, it’s not possible for us to procure this artefact unless the vulnerability is not detected by SSAP. Could you include the fix for this on your side instead?

Appreciate it.

**Comment by cerveada** [comment]
I think best solution here (since it's false positive) would be to open a ticket on SSAP to fix this. @jermynyeo could you please open the ticket? Or provide a link where one can open such ticket.

**Comment by wajda** [comment]
> Hello, I just checked, it’s not possible for us to procure this artefact unless the vulnerability is not detected by SSAP. Could you include the fix for this on your side instead?
> 
> Appreciate it.

@jermynyeo can you specify exactly which artifact are you referring to and what SSAP warning is?
As explained above I don't quite understand what kind of "fix" we can place to make SSAP satisfied. I tend to agree with @cerveada that SSAP false alarm is not our problem

**Comment by wajda** [comment]
Another question is - doesn't Spark, Kafka or other related libraries trigger the same alarm?
Since that Jackson dependency comes directly from Spark, Kafka and is not used in Spline at all, why would one blame Spline but not the source libraries?

**Comment by wajda** [comment]
I tried to explicitly exclude  `jackson-databind` dependency from the project, or mark it as _provided_ to make it more obvious for security scanners that the project does not directly depend on this library.  But unfortunately all the ways I tried had implications on the _test_ dependency tree where the jackson _is_ used transitively from Spark.
So, my final conclusion for this issue would be to close it as irrelevant.
